controllers/gateway_controller_reconciler_utils.go

```diff
@@ -152,8 +152,12 @@ func (r *GatewayReconciler) getGatewayConfigForGatewayClass(ctx context.Context,
 	}
 
 	gatewayConfig := new(operatorv1alpha1.GatewayConfiguration)
+	var namespace string
+	if gatewayClass.Spec.ParametersRef.Namespace != nil {
```

Comment:
For thorough-ness we should check if `ParametersRef` is `nil`:

```suggestion
	if gatewayClass.Spec.ParametersRef != nil && gatewayClass.Spec.ParametersRef.Namespace != nil {
```

Reply:
I see :thinking: 

So functionally that works for our current purposes but it means that the functionality of `getGatewayConfigForGatewayClass` doesn't have this problem solved in encapsulation (that is to say, by itself in a vacuum) so if other callers were to start calling this method it may not be clear to them that this problem exists, and that could create some potential for bugs for any new callers down the road that wouldn't be abundantly clear to the caller unless they carefully studied other calling code.

Generally we shouldn't have to study the calling code for a function or method to understand how to use it properly: we should aim for the function signature and godoc definition to give us as much as possible and for these kinds of problems to be solved internally so I would still advocate that we make this change here for completeness of the method.

Reply:
I get it, I'll add the check also here.